engine/crates/tracing/src/otel/metrics.rs

```diff
@@ -0,0 +1,79 @@
+use opentelemetry_sdk::metrics::data::Temporality;
+use opentelemetry_sdk::metrics::reader::{AggregationSelector, TemporalitySelector};
+use opentelemetry_sdk::metrics::{Aggregation, InstrumentKind, PeriodicReader, SdkMeterProvider};
+use opentelemetry_sdk::runtime::Runtime;
+use opentelemetry_sdk::Resource;
+use std::time::Duration;
+
+use crate::config::TracingConfig;
+use crate::error::TracingError;
+
+pub(super) fn build_meter_provider<R>(
+    runtime: R,
+    config: &TracingConfig,
+    resource: Resource,
+) -> Result<SdkMeterProvider, TracingError>
+where
+    R: Runtime,
+{
+    let mut provider = SdkMeterProvider::builder().with_resource(resource);
+
+    if config
+        .exporters
+        .stdout
+        .as_ref()
+        .map(|cfg| cfg.enabled)
+        .unwrap_or_default()
+    {
+        let reader = PeriodicReader::builder(opentelemetry_stdout::MetricsExporter::default(), runtime.clone())
+            .with_interval(Duration::from_secs(10))
+            .with_timeout(Duration::from_secs(10))
+            .build();
+
+        provider = provider.with_reader(reader);
+    }
+
+    #[cfg(feature = "otlp")]
+    if let Some(config) = config.exporters.otlp.as_ref().filter(|cfg| cfg.enabled) {
+        use opentelemetry_otlp::MetricsExporterBuilder;
+        let exporter = crate::otel::provider::build_otlp_exporter::<MetricsExporterBuilder>(config)?
```

Comment:
I'd move the `build_otlp_exporter` to a separate mod (`exporter`?) and rename the existing `provider` to `tracing`.

Reply:
done, renaming to `traces` rather than `tracing` though